src/serialization/UntypedDecimal.cs

```diff
@@ -0,0 +1,20 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Kiota.Abstractions.Serialization
+{
+    /// <summary>
+    /// Represents an untyped node with decimal value.
+    /// </summary>
+    /// <param name="value">The decimal value associated with the node.</param>
+    public class UntypedDecimal(decimal value) : UntypedNode
+    {
+        private readonly decimal _value = value;
+        /// <summary>
+        /// Gets the value associated with untyped decimal node.
+        /// </summary>
+        /// <returns>The value associated with untyped decimal node.</returns>
+        public override object GetValue() => _value;
```

Comment:
I initially expected a change like this. But this seems to be only supported in C# 9(NET5 and above)
```suggestion
        public override decimal GetValue() => _value;
```

What I'd suggest is to change to below to avoid the casting of the return type as done in the json library PR.
```suggestion
        public new decimal GetValue() => _value;
```

Just to loop in @baywet . 
Moving from a property to a method should work in the various languages which should support some form of return type covariance. 
- [TS ](https://stackoverflow.com/questions/62515165/method-overriding-in-typescript-with-different-return-types-without-hack)
- [Java](https://www.geeksforgeeks.org/covariant-return-types-java/)
- [PHP](https://www.geeksforgeeks.org/covariance-and-contravariance-in-php/)
- [C#](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/covariant-returns#motivation) (though we may use method hiding till we drop netstandard support)
- [Go](https://go.dev/play/p/fsK0Gi_r4eC)
- [Python](https://mypy.readthedocs.io/en/latest/generics.html#type-variables-with-upper-bounds)

Reply:
Thanks for doing the research work here, it helps!
I'd like us to have early implementations for Go/Java/TS before we proceed with merging the dotnet ones if you don't mind?

Reply:
Thats totally cool. Thanks for taking a look!